stingray/powerspectrum.py

```diff
@@ -386,3 +386,44 @@ def _make_segment_spectrum(self, lc, segment_size):
             end_ind += nbins
 
         return power_all, nphots_all
+
+
+class DynamicalPowerspectrum(AveragedPowerspectrum):
+    def __init__(self, lc, segment_size, norm="frac"):
+        """
+        Parameters
+        ----------
+        lc : lightcurve.Lightcurve object
+            The time series of which the Dynamical powerspectrum is
+            to be calculated.
+
+        segment_size : float, default 1
+             Length of the segment of light curve, default value is 1 second.
+
+        norm: {"leahy" | "frac" | "abs" | "none" }, optional, default "frac"
+            The normaliation of the periodogram to be used. Options are
+            "leahy", "frac", "abs" and "none", default is "frac".
+
+        Attributes
+        ----------
+        dyn_ps : np.ndarray
+            The matrix with fourier frequencies in the first column and time
+            in the second column.
+        """
+        if segment_size < 2*lc.dt:
+            raise ValueError("Length of the segment is too short to form a "
+                             "light curve!")
+        elif segment_size > lc.tseg:
+            raise ValueError("Length of the segment is too long to create "
+                             "any segments of the light curve!")
+        self.segment_size = segment_size
+        self.norm = norm
+        power_all, _ = AveragedPowerspectrum._make_segment_spectrum(
+            self, lc, segment_size)
+        self._make_matrix(power_all)
+
+    def _make_matrix(self, power_all):
+        """Create the matrix with freq and time columns."""
+        row1 = power_all[0].freq
+        row2 = list(range(0, self.segment_size*len(row1), self.segment_size))
```

Comment:
@OrkoHunter I think @matteobachetti  is right, `segment_size` should be able to be a non integer amount of time. That can be done by using `np.arange()` instead of `range()`.

I think the test to be made is `segment_size *  number_of_segments <= lc.tseg` And if it's not true, either throw away the last segment (`number_of_segments -= 1`) or change `segment_size` only for the last segment.